README.md

```diff
@@ -229,6 +230,11 @@ x-transportd:
   strip:
     # (int) Number of URL segments to remove from the beginning of the path before redirect.
     count: 0
+  headerinject:
```

Comment:
Question: Are headers mapped to value by index in each list? It seems like it'd be more readable/clearer to have a list of name/value pairs, e.g.:

```
headerinject:
      (string) Header name to inject
    - name: header1
       value: value1
    - name: header2
       value: value2
```

Reply:
I intentionally left `map[string]*` support out of `settings` to prevent cases where we might use them, intentionally or not, to avoid the more strict structure of a struct. It's not out of the question to add support for this but one question would be: "How would you represent a map[string]something as an environment variable value?". The `settings` project uses https://github.com/spf13/cast which would expect a JSON formatted entry like `HEADER_INJECT_VALUES='{"header1": "value1"}'` which is not that intuitive.

Another option here that I'm not too fond of is having the options be a string slice such as:

```yaml
headerinject:
    headers:
        - header1:value1
        - header2:value2
```

I'm not happy about introducing a special format because it would then require special parsing to get the values and I'd rather rely on the `settings` type system to parse things for us.

Reply:
I'm just worried that this format might expose users to off-by-one errors, but the other solutions don't sound much better to me.